examples/connected-car-cluster-sharding/akka-connected-car-streamlet/src/main/scala/connectedcar/streamlets/ConnectedCarCluster.scala

```diff
@@ -1,37 +1,59 @@
 package connectedcar.streamlets
 
-import akka.actor.{ ActorRef, Props }
-import akka.cluster.sharding.{ ClusterSharding, ClusterShardingSettings }
+import akka.actor.{ActorRef, Props}
+import akka.cluster.sharding.typed.scaladsl.{ClusterSharding, Entity, EntityRef, EntityTypeKey}
 import akka.util.Timeout
-import cloudflow.akkastream.scaladsl.{ FlowWithCommittableContext, RunnableGraphStreamletLogic }
+import akka.actor.typed.scaladsl.adapter._
+import akka.cluster.sharding.external.ExternalShardAllocationStrategy
+import akka.cluster.sharding.typed.ClusterShardingSettings
+import cloudflow.akkastream.scaladsl.{FlowWithCommittableContext, RunnableGraphStreamletLogic}
 import cloudflow.streamlets.StreamletShape
-import cloudflow.streamlets.avro.{ AvroInlet, AvroOutlet }
-import connectedcar.actors.ConnectedCarActor
+import cloudflow.streamlets.avro.{AvroInlet, AvroOutlet}
+import connectedcar.actors.{ConnectedCarActor, ConnectedCarERecordWrapper}
 
 import scala.concurrent.duration._
 import akka.pattern.ask
-import cloudflow.akkastream.{ AkkaStreamlet, Clustering }
-import connectedcar.data.{ ConnectedCarAgg, ConnectedCarERecord }
+import akka.stream.scaladsl.Sink
+import cloudflow.akkastream.{AkkaStreamlet, Clustering}
+import connectedcar.data.{ConnectedCarAgg, ConnectedCarERecord}
 
 object ConnectedCarCluster extends AkkaStreamlet with Clustering {
   val in    = AvroInlet[ConnectedCarERecord]("in")
   val out   = AvroOutlet[ConnectedCarAgg]("out", m ⇒ m.driver.toString)
   val shape = StreamletShape(in).withOutlets(out)
 
   override def createLogic = new RunnableGraphStreamletLogic() {
-    def runnableGraph = sourceWithCommittableContext(in).via(flow).to(committableSink(out))
+    val typeKey = EntityTypeKey[ConnectedCarERecordWrapper]("Car")
 
-    val carRegion: ActorRef = ClusterSharding(context.system).start(
-      typeName = "Counter",
-      entityProps = Props[ConnectedCarActor],
-      settings = ClusterShardingSettings(context.system),
-      extractEntityId = ConnectedCarActor.extractEntityId,
-      extractShardId = ConnectedCarActor.extractShardId
-    )
+    val source = shardedSourceWithCommittableContext(in,
+                                      typeKey: EntityTypeKey[ConnectedCarERecordWrapper],
+                                      (msg: ConnectedCarERecordWrapper) => msg.record.carId+""
+                                    )
+
+    val clusterSharding = ClusterSharding(system.toTyped)
```

Comment:
This could be added to AkkaStreamletContext ?

Reply:
Considering `clusterSharding` wouldn't be created for every AkkaStremlet, only the ones that use `with Clustering`, I don't think it makes sense to add the def to AkkaStreamletLogic.  Users can always create it for themselves if they need it.  

Is there any reason ClusterSharding shouldn't be called multiple times? If not I think it's fine to keep it as is.

Reply:
I thought of it as a convenience method so you don't have to do `ClusterSharding(system.toTyped)` but just create it with a `def clusterSharding` that is available.

Reply:
ahh that makes sense